### DISCUSSION TITLE
Skynet 框架中的 mongo 模块已经太老了，是否考虑更新一下？

### DISCUSSION THREAD

**Discussion by aceyin** [cloudwu/skynet#1873] [General]
比如 在 `skynet/mongo.lua` 中于 `auth` 相关的一些代码，里面用到了 mongo 的 `getnonce` 等命令。

但这些命令已经被 mongo 官方标记为 `deprecated` 了：[getnonce](https://www.mongodb.com/docs/v4.4/reference/command/getnonce/)。

看看是否有计划更新一版 `mongo` 模块？

**Comment by cloudwu** [comment]
有人有兴趣就可以改。

我没有兴趣。

**Comment by cloudwu** [comment]
我看了一下。getnonce 是 mongodb-cr 的一部分。

而 mongodb-cr 已经 deprecated 。

https://www.mongodb.com/docs/drivers/node/current/fundamentals/authentication/mechanisms/#mongodb-cr

所以，直接不用 mongodb-cr 即可。就这个问题，应该什么都不用做。而我没看到别的什么需要更新的。

**Reply by tzngit** [reply]
比如支持SRV连接呢？ 支持TLS呢？ 因为现在很多云服务商的sharding集群都要求客户端支持SRV了，如果skynet能支持就好了

**Reply by sniper00** [reply]
> 比如支持SRV连接呢？ 支持TLS呢？ 因为现在很多云服务商的sharding集群都要求客户端支持SRV了，如果skynet能支持就好了

可以考虑利用rust生态 https://github.com/sniper00/lrust?tab=readme-ov-file#4-mongodb

**Reply by tzngit** [reply]
> > 比如支持SRV连接呢？ 支持TLS呢？ 因为现在很多云服务商的sharding集群都要求客户端支持SRV了，如果skynet能支持就好了
> 
> 可以考虑利用rust生态 https://github.com/sniper00/lrust?tab=readme-ov-file#4-mongodb

可惜不行啊，你这个只是一个driver， 并没有集成进skynet，支持SRV、TLS的mongo driver官方也有很多，比如mongo-c-driver，但是关键是集成进skynet，尤其是网络层必须用skynet的，这个才是重点

**Reply by sniper00** [reply]
> > > 比如支持SRV连接呢？ 支持TLS呢？ 因为现在很多云服务商的sharding集群都要求客户端支持SRV了，如果skynet能支持就好了
> > 
> > 
> > 可以考虑利用rust生态 https://github.com/sniper00/lrust?tab=readme-ov-file#4-mongodb
> 
> 可惜不行啊，你这个只是一个driver， 并没有集成进skynet，支持SRV、TLS的mongo driver官方也有很多，比如mongo-c-driver，但是关键是集成进skynet，尤其是网络层必须用skynet的，这个才是重点

网络层用skynet不是必须的，我就是用的 tokio的异步网络层，已经集成到我的框架中 https://github.com/sniper00/moon/blob/master/src/moon/main.cpp#L292
https://github.com/sniper00/lrust/blob/main/crates/libs/lib-lualib/src/lib.rs#L18 
你可以参考集成到skynet中

**Reply by tzngit** [reply]
不知道moon具体是什么，看起来是rust版本的类似skynet的东西， 但是显然，skynet有自己的网络层，因为一个driver而引入tokio是不合适的，这样skynet里会有两个网络层实现....
可能最合适的方法就是看懂官方driver的c代码，然后继续在现有skynet mongo drvier的lua或者c代码里实现同样的东西

**Reply by sniper00** [reply]
> 不知道moon具体是什么，看起来是rust版本的类似skynet的东西， 但是显然，skynet有自己的网络层，因为一个driver而引入tokio是不合适的，这样skynet里会有两个网络层实现.... 可能最合适的方法就是看懂官方driver的c代码，然后继续在现有skynet mongo drvier的lua或者c代码里实现同样的东西

底层都是调用 epoll之类的，使用rust版的数据库client比较便捷，无需过多维护(如果你能自己改skynet的driver另说)，并且性能比skynet 实现的要高。
